src/pipe_windows.rs

```diff
@@ -0,0 +1,204 @@
+//! up even when no signal came after the signal was taken, because ‒ it arrived between cleaning
+//! the pipe and taking the action). Note that some OS primitives (eg. `select`) suffer from
+//! spurious wakeups themselves (they can claim a FD is readable when it is not true) and blocking
+//! `read` might return prematurely (with eg. `EINTR`).
+//!
+//! The reverse order of first taking the action and then clearing the pipe might lose signals,
+//! which is usually worse.
+//!
+//! This is not a problem with blocking on reading from the pipe (because both the blocking and
+//! cleaning is the same action), but in case of asynchronous handling it matters.
+//!
+//! If you want to combine setting some flags with a self-pipe pattern, the flag needs to be set
+//! first, then the pipe written. On the read end, first the pipe needs to be cleaned, then the
+//! flag and then the action taken. This is what the [`Signals`](../iterator/struct.Signals.html)
+//! structure does internally.
+//!
+//! # Write collating
+//!
+//! While unlikely if handled correctly, it is possible the write end is full when a signal comes.
+//! In such case the signal handler simply does nothing. If the write end is full, the read end is
+//! readable and therefore will wake up. On the other hand, blocking in the signal handler would
+//! definitely be a bad idea.
+//!
+//! However, this also means the number of bytes read from the end might be lower than the number
+//! of signals that arrived. This should not generally be a problem, since the OS already collates
+//! signals of the same kind together.
+//!
+//! # Examples
+//!
+//! This example waits for at last one `SIGUSR1` signal to come before continuing (and
+//! terminating). It sends the signal to itself, so it correctly terminates.
+//!
+//! ```rust
+//! extern crate libc;
+//! extern crate winapi;
+//! extern crate signal_hook;
+//!
+//! use std::fs::File;
+//! use std::io::{Error, Read};
+//! use std::ptr;
+//! use std::os::windows::prelude::*;
+//! use winapi::um::namedpipeapi::CreatePipe;
+//! use winapi::um::winnt::HANDLE;
+//!
+//! fn main() -> Result<(), Error> {
+//!     let (mut read, write) = {
+//!         let mut read: HANDLE = ptr::null_mut();
+//!         let mut write: HANDLE = ptr::null_mut();
+//!         let success = unsafe { CreatePipe(&mut read, &mut write, ptr::null_mut(), 1024) } != 0;
+//!         assert!(success);
+//!         let read = unsafe { File::from_raw_handle(read as RawHandle) };
+//!         let write = unsafe { File::from_raw_handle(write as RawHandle) };
+//!         (read, write)
+//!     };
+//!     signal_hook::pipe_windows::register_handle(signal_hook::SIGINT, write)?;
+//! # fn do_something_that_may_be_interrupted_by_ctrl_c() {}
+//!     do_something_that_may_be_interrupted_by_ctrl_c();
+//!     let mut buff = [0];
+//! # if false {
+//!     read.read_exact(&mut buff)?;
+//! # }
+//!     println!("Happily terminating");
+//!     Ok(())
+//! }
+//! ```
```

Comment:
This seems the documentation here talks mostly about unix too ‒ there are things like `UnixStream` mentioned.